stl/inc/chrono

```diff
@@ -609,8 +610,22 @@ namespace chrono {
         static constexpr bool is_steady = true;
 
         _NODISCARD static time_point now() noexcept { // get current time
-            const long long _Freq = _Query_perf_frequency(); // doesn't change after system boot
-            const long long _Ctr  = _Query_perf_counter();
+#if (defined(_M_ARM) || defined(_M_IX86) || defined(_M_ARM64) || defined(_M_X64)) && !defined(_M_CEE_PURE)
+            // ad-hoc <atomic> load/compare_exchange_strong with std::memory_order_relaxed
+            // avoid dependency on real <atomic>
+            static volatile long long _Freq_cached = -1; // doesn't change after system boot
+            // _Freq_cached must always be accessed via an Interlocked function,
+            // But it is accessed via __iso_volatile_load64, which is also proper way
+#pragma warning(suppress : 28112)
+            long long _Freq = __iso_volatile_load64(&_Freq_cached);
+            if (_Freq == -1) {
+                _Freq = _Query_perf_frequency();
+                _InterlockedCompareExchange64(&_Freq_cached, _Freq, -1);

```

Comment:
I think this should be _InterlockedExchange64 -- no value in the compare here. Even better would be to mimic what atomic does for a relaxed store since there are no memory ordering concerns here.

Reply:
InterlockedCompareExchange mimics atomic store on x86, even relaxed. There's no 64-bit exchange for x86. Others are the same for unification, since this operation is not frequent anyway (let me know if you prefer #ifdef for every case instead).

I've tried __iso_volatile_store64, but it was causing ICE.

Reply:
On x86 yes, there would be no benefit. I guess it isn't a huge deal because this path is unlikely but if it isn't doing exactly what `<atomic>` does there will always be questions as to why. If we're getting ICEs, we need to reduce / report those to the compiler team, not silently work around them. On which compiler did you observe that?

Reply:
@cbezault  told in a comment above the revision https://github.com/microsoft/STL/pull/646/commits/8ce02679dbe14899135aa9d76aacf13561096fa5 causes ICE on x86.

I assume it one of tests compilation.

I've seen how `__iso_volatile_store64` compiles. It uses SSE/AVX registers. With `/arch:IA32` it is based on FPU `fild` / `fistp` instructions. So it looks beneficial over what's `atomic_int64` is doing now.

On the other hand, it is not documented, and does not compile in non-preview Visual Studio, so maybe it does not yet exist. Then ICE is no wonder.

Reply:
I've put the finding on `__iso_volatile_store64` / ICE to #651, so that it can be tracked separately from this PR